openmmtools/integrators.py

```diff
@@ -1185,25 +1094,536 @@ def substep_function(step_string):
+        elif step_string == "H":
+            self.addAlchemicalPerturbationStep()
+
+    def addGlobalVariables(self, nsteps):
+        """
+        Add the appropriate global parameters to the CustomIntegrator. nsteps refers to the number of
+        total steps in the protocol.
+
+        Parameters
+        ----------
+        nsteps : int, greater than 0
+            The number of steps in the switching protocol.
+        """
+        self.addGlobalVariable('lambda', 0.0) # parameter switched from 0 <--> 1 during course of integrating internal 'nsteps' of dynamics
+        self.addGlobalVariable('kinetic', 0.0) # kinetic energy
+        self.addGlobalVariable('nsteps', nsteps) # total number of NCMC steps to perform
+        self.addGlobalVariable('step', 0) # current NCMC step number
+
+class ExternalPerturbationLangevinSplittingIntegrator(LangevinSplittingIntegrator):
+    """
+    LangevinSplittingIntegrator that accounts for external perturbations and tracks protocol work.
+    """
+
+    def __init__(self,
+                 splitting="V R O R V",
+                 temperature=298.0 * simtk.unit.kelvin,
+                 collision_rate=1.0 / simtk.unit.picoseconds,
+                 timestep=1.0 * simtk.unit.femtoseconds,
+                 constraint_tolerance=1e-8,
+                 measure_shadow_work=False,
+                 measure_heat=True):
+
+        self.addGlobalVariable("protocol_work", 0)
+        self.addGlobalVariable("perturbed_pe", 0)
+        self.addGlobalVariable("unperturbed_pe", 0)
+        super(ExternalPerturbationLangevinSplittingIntegrator, self).__init__(splitting=splitting,
+                                                                              temperature=temperature,
+                                                                              collision_rate=collision_rate,
+                                                                              timestep=timestep,
+                                                                              constraint_tolerance=constraint_tolerance,
+                                                                              measure_shadow_work=measure_shadow_work,
+                                                                              measure_heat=measure_heat)
+
+    def add_integrator_steps(self, splitting, measure_shadow_work, measure_heat, ORV_counts, force_group_nV, mts):
+        self.addComputeGlobal("perturbed_pe", "energy")
+        self.addComputeGlobal("protocol_work", "protocol_work + (perturbed_pe - unperturbed_pe)")
+        super(ExternalPerturbationLangevinSplittingIntegrator, self).add_integrator_steps(splitting, measure_shadow_work, measure_heat, ORV_counts, force_group_nV, mts)
+        self.addComputeGlobal("unperturbed_pe", "energy")
 
 class VVVRIntegrator(LangevinSplittingIntegrator):
     """Create a velocity Verlet with velocity randomization (VVVR) integrator."""
```

Comment:
Be sure to remove the measure_protocol work flag. 

Also, under test_integrators.py, for the `test_vvvr_protocol_work_accumulation`  you will need to change the code to use the `ExternalPerturbationLangevinSplittingIntegrator` integrator (and probably just initialize it with the VVVR splitting).